### ISSUE TITLE
ASSERT(0) in docnode.cpp

### ISSUE THREAD

**Issue by kjellahl** [doxygen/doxygen#10950]
Starting with doxygen 1.9.3 the ASSERT(0) statement in docnode.cpp, DocPara::handleCommand()
can be reached. The attached doxytest.h file is an example of a file that triggers
this message. It's part of a longer file that produces several ASSERT messages.

I made some tests with `gdb` and saw that cmdId = 103 (CMD_SETSCOPE) which
is not a `case` in the big `switch` statement.
Tested with doxygen 1.12.0 (7b32bc31631193b9801702b2fb643ee51dcbc981).

[doxytest.zip](https://github.com/user-attachments/files/15888016/doxytest.zip)

**Comment by albert-github** [comment]
The probem in this case looks like to be the markdown conversion from `>` to `<blockquote>`, more precise the detection of the end of the range, seen:
```
======== Markdown =========
---- input -------
The action that this activatable will activate and receive
 updates from for various states and possibly appearance.

 > Gtk::Activatable implementors need to handle the this property and
 > call Gtk::Activatable::do_set_related_action() when it changes.

 @deprecated Use Gtk::Actionable::property_action_name() on Gtk::Actionable widgets instead.

---- output -----
The action that this activatable will activate and receive
 updates from for various states and possibly appearance.

<blockquote>&zwj;Gtk::Activatable implementors need to handle the this property and
call Gtk::Activatable::do_set_related_action() when it changes.

 @deprecated Use Gtk::Actionable::property_action_name() on Gtk::Actionable widgets instead.
</blockquote>
=========
```

**Comment by doxygen** [comment]
@kjellahl Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by kjellahl** [comment]
Yes, the 7673d22af1cea4d84bdc00164d9c9a25d693a714 commit fixes the problem.

**Comment by doxygen** [comment]
@albert-github Pushed a fix for the underlying issue. Doxygen will now report `warning: found </blockquote> tag without matching <blockquote>`  inside `deprecated`, because the end of the block quote is taken as part of the `@deprecated` command (this command ends at a new paragraph).

**Comment by albert-github** [comment]
@doxygen 
- I saw that part of the commit was reverted, so is it still ongoing work? I still get the warnings:
  ```
  deprecated:4: warning: Unexpected tag <dt> found
  deprecated:5: warning: Unexpected tag <dd> found
  ``` 
  Note I don't think the initial warning `found </blockquote> tag without matching <blockquote>` wouldn't be nice either.
- the problem was specified for the `<blockquote>` tag but it is not limited to this, e.g. `<details>` also has this problem, though it looks like this gives after the "partly reverted commit" the same messages as `<blockquote>`.

**Comment by doxygen** [comment]
@albert-github The fix revealed an existing problem in the markdown processing (extra end of blockquote) which now triggered a warning and caused the manual generation step of the build to fail. I fixed this now. Also included the details command.

**Comment by albert-github** [comment]
Looks like that there are more HTML / XML tags that cause problems in relation to `\deprecated` etc.
E.g. `<code>` where the code is ended by the `\xrefitem ` command analogous for the `<pre>` (and probably more, didn't investigate yet. Probably `<blockquote>` / `<details>` / `<code>` / `<pre>` should be solved first before further testing).

**Comment by doxygen** [comment]
@albert-github If you put an HTML end tag inside a `@deprecated` command, it is not strange you get an error.
If one formats it a bit different you'll see the problem a bit more clearly

```c++
  /*
   * <blockquote>
   * Gtk::Activatable implementors need to handle the this property and
   * call Gtk::Activatable::do_set_related_action() when it changes.
   *
   * @deprecated Use Gtk::Actionable::property_action_name() on Gtk::Actionable widgets instead. </blockquote>
   * 
   * missing end blockquote...
   */
```

It is similar to incorrectly nesting commands

```c++
/** <blockquote>
  * <ul>
  * <li>Some text </blockquote>
  * </ul>
  */
```

**Comment by albert-github** [comment]
It is quite complicated .
The second example is obvious as here the nesting of the commands `<ul>` and `<blockquote>` though when removing the `</ul>` out like:
```
/** \file
  * <blockquote>
  * <ul>
  * <li>Some text </blockquote>
  */
```
we don't get a warning, whilst when also removing the `</blockquote>` like
```
/** \file
  * <blockquote>
  * <ul>
  * <li>Some text
  */
```
we get a warning like:
```
dd.h:4: warning: unexpected end of comment while inside <ul> block
```

Example: [example.tar.gz](https://github.com/user-attachments/files/16044118/example.tar.gz)

**Comment by doxygen** [comment]
I think the more structural solution is to treat certain HTML end tags such as `</blockquote>`, `</details>` and `</div>` as explicitly also ending the current paragraph. This way the warnings can also be improved. Maybe better to file this as a different issue, as it is not really related with the `ASSERT(0)` of the original issue.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by kjellahl** [comment]
Yes, this issue is fixed in doxygen 1.12.0.
